pkg/tokenvalidation/bootstrapauthenticator.go

```diff
@@ -0,0 +1,102 @@
+package oauth
+
+import (
+	"context"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	kauthenticator "k8s.io/apiserver/pkg/authentication/authenticator"
+	kuser "k8s.io/apiserver/pkg/authentication/user"
+
+	authorizationv1 "github.com/openshift/api/authorization/v1"
+	userv1 "github.com/openshift/api/user/v1"
+	oauthclient "github.com/openshift/client-go/oauth/clientset/versioned/typed/oauth/v1"
+	bootstrap "github.com/openshift/library-go/pkg/authentication/bootstrapauthenticator"
+)
+
+const ClusterAdminGroup = "system:cluster-admins"
+
+type bootstrapAuthenticator struct {
+	tokens            oauthclient.OAuthAccessTokenInterface
+	getter            bootstrap.BootstrapUserDataGetter
+	validator         OAuthTokenValidator
+	implicitAudiences kauthenticator.Audiences
+}
+
+func NewBootstrapAuthenticator(tokens oauthclient.OAuthAccessTokenInterface, getter bootstrap.BootstrapUserDataGetter, implicitAudiences kauthenticator.Audiences, validators ...OAuthTokenValidator) kauthenticator.Token {
```

Comment:
what the purpose of this authenticator?

Reply:
The purpose of this whole commit is to pull the authenticators from o/k to simplify review by showing how they changed. This specific authenticator gets merged into the main tokenvalidationhandler (https://github.com/openshift/oauth-apiserver/pull/9/commits/cfd5628e085a590e6d0eea0befdc94f327cf3605#diff-3e4e445b703e99b73d48bdb579ec3e69502289fa4245fef482ba16f91ab0d0eeR115) so that we finally stop making development mistakes by forgetting about this special cased authenticator.

Its original purpose was to authenticate the kube:admin user.